device-tree/src/fdt.rs

```diff
@@ -61,6 +61,11 @@ impl<'a> Fdt<'a> {
         self.get_module_node_region("multiboot,ramdisk")
     }
 
+    /// Returns an iterator over CPU nodes.
+    pub fn cpus(&self) -> CpuIter<'_, 'a> {
```

Comment:
Does this work?
```
    /// Returns an iterator over CPU nodes.
    pub fn cpus(&self) -> impl Iterator<Item = Cpu<'_, 'a>> {
        self.inner
            .nodes()
            .filter(|n| {
                n.props()
                    .any(|p| Ok(p.name()? == "device_type" && p.str()? == "cpu"))
            })
            .iterator()
            .map_while(|r| r.map(|n| Cpu { inner: n }).ok())
    }
```

Then the CpuIter below isn't needed. I realize the fallible -> normal iter conversion is messy and the other iters here don't set that example(but can, I think), so I'm OK either way.

Reply:
Ah, thanks for calling out the `iterator()` function! Replaced now with something along the lines of your suggestion.